api/v1beta1/azureclusteridentity_webhook.go

```diff
@@ -26,3 +30,31 @@ func (c *AzureClusterIdentity) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		For(c).
 		Complete()
 }
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1beta1-azureclusteridentity,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azureclusteridentities,versions=v1beta1,name=validation.azureclusteridentity.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
```

Comment:
I am not sure but do we want to restrict the `webhook:verbs` to just `create` because we do not have any logic for `ValidateUpdate()`?

Reply:
I am not sure if a Kubernetes API gives out an `UPDATE` call while creating the resource(i.e. at the time of object creation). Or to put it other way, I am not sure if Kubernetes API's UPDATE calls are really needed here for `ValidateCreate()` calls.

Reply:
The update webhook is there so the AzureClusterIdentity webhook can fully implement `webhook.Validator`. The update is a no-op right now because we were only concerned with blocking the creating of an AzureClusterIdentity. I'll try and observe the update behavior and make sure it isn't doing something unintended.

Reply:
Oh sorry, I was pointing to the `update` verb in the `+kubebuilder:webhook:verbs=create;update` marker and not to the implementation of `ValidateUpdate()` function. 

I found the following upon investigation (referring from [Kubebuilder webhook page](https://book.kubebuilder.io/reference/markers/webhook.html))
```
verbs=<[]>string
specifies the Kubernetes API verbs that this webhook receives requests for.
Only modification-like verbs may be specified. May be “create“, “update“, “delete“, “connect“, or “*“ (for all).
```


I am not sure if the `verbs=update` is required here in the marker.

Reply:
Ah good catch! That should be removed, I'll make a change now.